product_contract/models/product_template.py

```diff
@@ -167,3 +167,25 @@ def _check_contract_product_type(self):
         """
         if any([product.is_contract and product.type != "service" for product in self]):
             raise ValidationError(_("Contract product should be service type"))
+
+    @api.constrains("auto_renew_interval")
+    def _check_auto_renew_interval_is_strictly_positive(self):
+        for product in self:
+            if not product.is_contract or not product.is_auto_renew:
+                return
+            if product.auto_renew_interval <= 0:
+                raise ValidationError(
+                    f"Value of {product._fields['auto_renew_interval'].string}"
```

Comment:
You want to make your error message translatable by wrapping them in `env._` .

Be aware that vars included in the string will be evaluated inside the source string to translate.

https://www.odoo.com/documentation/18.0/developer/howtos/translations.html


What you want should be pretty close to this:

https://github.com/odoo/odoo/blob/18.0/addons/web_editor/models/ir_ui_view.py#L84-L88